src/control/test/pp_test.cpp

```diff
@@ -113,4 +116,161 @@ TEST(PurePursuitTests, Test_pp_steering_control_law_1) {
   //  Alpha: 0.804189
   //  ld_: 5.38516
   EXPECT_NEAR(0.381, steering_cmd, 0.001);
-}
\ No newline at end of file
+}
+
+/**
+ * @brief Test PurePursuit - update_closest_point()
+ */
+TEST(PurePursuitTests, Test_update_closest_point_1) {
+  std::ifstream trackFile("../../src/control/test/assets/track1.csv");
+  ASSERT_TRUE(trackFile.is_open()) << "Failed to open track file";
+
+  custom_interfaces::msg::PathPointArray path_msg;
+
+  std::string line;
+
+  std::getline(trackFile, line); // Skip the first line
+
+  while (std::getline(trackFile, line)) {
+    std::istringstream iss(line);
+    std::string x, y, v;
+    if (std::getline(iss, x, ',') && std::getline(iss, y, ',') && std::getline(iss, v)) {
+      // Create a PathPoint and add it to the PathPointArray
+      custom_interfaces::msg::PathPoint point;
+      point.x = std::stod(x); //string to double
+      point.y = std::stod(y);
+      point.v = std::stod(v);
+      path_msg.pathpoint_array.push_back(point);
+    }
+  }
+  trackFile.close();
+
+  // Convert path_msg to ConstSharedPtr
+  const custom_interfaces::msg::PathPointArray::ConstSharedPtr path_msg_ptr =
+      std::make_shared<const custom_interfaces::msg::PathPointArray>(path_msg);
+
+
+  double k = 0.5;
+  double ld_margin = 0.1;
+  PurePursuit pp(k, ld_margin);
+  pp.vehicle_pose_.rear_axis_ = Point(47.0, -13.0);
+  Point expected_point = Point(46.5, -12.37); 
+  int expected_id = 76;
+
+  auto result = pp.update_closest_point(path_msg_ptr);
+  
+  EXPECT_EQ(result.first.x_, expected_point.x_);
+  EXPECT_EQ(result.first.y_, expected_point.y_);
+  EXPECT_EQ(result.second, expected_id);
+}
+
+/**
+ * @brief Test PurePursuit - update_lookahead_point()
+ */
+
+TEST(PurePursuitTests, Test_update_lookahead_point_1) {
+  std::ifstream trackFile("../../src/control/test/assets/track1.csv");
+  ASSERT_TRUE(trackFile.is_open()) << "Failed to open track file";
+
+  custom_interfaces::msg::PathPointArray path_msg;
+
+  std::string line;
+
+  std::getline(trackFile, line); // Skip the first line
+
+  while (std::getline(trackFile, line)) {
```

Comment:
This part of reading the file can probably be extracted to a function to not repeat itself so much